packages/blade/src/components/Input/OTPInput/_decisions/_decisions.md

```diff
@@ -0,0 +1,92 @@
+# OTPInput Decisions <!-- omit in toc -->
+
+This doc talks about the API decisions for `OTPInput`. 
+## Index <!-- omit in toc -->
+
+- [OTPInput API](#otpinput-api)
+  - [Sample Usage](#sample-usage)
+- [Keyboard return key types for web and native](#keyboard-return-key-types-for-web-and-native)
+- [OTP SMS auto-read feature](#otp-sms-auto-read-feature)
+  - [Web](#web)
+  - [Native Android](#native-android)
+  - [Native iOS](#native-ios)
+
+## OTPInput API
+
+<img src="./otp.png" width="300" alt="otp input" />
+
+### Sample Usage
+
+```tsx
+<OTPInput
+  otpLength={4}
+  label="Enter One Time Code"
+  labelPosition="left"
+  helpText="Enter the OTP sent to your phone number"
+  errorText="Invalid OTP"
+  successText="Valid OTP"
+  validationState={inputValue.length < 4 ? 'error' : 'success'}
+/>
+```
+
+
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| otpLength | `4`, `6`  | No | `4` | Determines the length of the OTP input. |
+| label | `string` | Yes |  | Determines the label of the input field. Also used as `aria-label` |
+| labelPosition | `top`, `left` | No | `top` | Used to identify if the label of the input field will be placed on the top of the input field or left. `left` is only available for large screen devices. When position is `left` then the width of the label is fixed to `120px` and the content is aligned left. |
+| necessityIndicator | `optional`, `required` | No |  | Indicator to be shown next to the label for emphasizing the necessity of the field. if `optional` is provided then it'll show `${label} (optional)` and if `required` is provided then it'll show `${label} *` |
```

Comment:
Similar to `PasswordField` can an OTP ever be optional?

Reply:
Based on the [OTP Input design guidelines](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=11534%3A151829), we should not have an `optional` tag, this should always be a mandatory field. Design needs to decide if we should have the `*` or not in the OTP Input field label irrespective.

Reply:
The indicator will depend on the design. We should give them the choice to show it or not

Reply:
On password field we're not allowing optional here, maybe we can follow the same convention here? Need to confirm with design.

Reply:
Discussed with Saurav, we won't be having any necessity indicator